### PR TITLE
Section on sending metrics to STF and Gnocchi

### PR DESCRIPTION
**Pull request by mrunge**
This section describes how to send metrics to both, STF and traditional telemetry.

### CODE REVIEW

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
+= Sending metrics to Gnocchi and to {ProjectShort}
+
+[role="_abstract"]
+
+In order to send metrics to STF and at the same time to gnocchi, the
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:5]:
```suggestion
To send metrics to {Project} ({ProjectShort}} and gnocchi simultaneously, you must include another configuration file to your deployment. 
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+following templates have to be added in addition to the templates
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:6]:
```suggestion
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+mentioned above.
+
+.Prerequisites
+
+An OpenStack environment file was created, see xref:configuring-the-stf-connection-for-the-overcloud[].
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:11]:
```suggestion
*  You have created a file that contains the connection configuration of the AMQ Interconnect for the overcloud to the {ProjectShort} deployment. For more information, see xref:configuring-the-stf-connection-for-the-overcloud[].
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+.Procedure
+
+. Create a configuration file named `publish-to-gnocchi.yaml` in the
+`/home/stack` directory.
+
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:17]:
```suggestion
+
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    CeilometerEnablePanko: false
+    GnocchiArchivePolicy: 'high'
+    GnocchiBackend: 'rbd'
+    GnocchiRbdPoolName: 'metrics'
+
+    EventPipelinePublishers: ['gnocchi://?filter_project=service&archive_policy=high']
+    PipelinePublishers: ['gnocchi://?filter_project=service&archive_policy=high']
+----
+
+The template should be added to the overcloud deploy command.
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:41]:
```suggestion
. Add the environment file `publish-to-gnocchi.yaml` to deployloyment command:
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:42]:
```suggestion
+
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+[source,bash,options="nowrap",subs="+quotes"]
+----
+openstack overcloud deploy <other arguments>
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:45]:
```suggestion
# openstack overcloud deploy <other arguments>
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  --templates /usr/share/openstack-tripleo-heat-templates \
+  --environment-file <...other-environment-files...> \
+  --environment-file /usr/share/openstack-tripleo-heat-templates/environments/metrics/ceilometer-write-qdr.yaml \
+  --environment-file /usr/share/openstack-tripleo-heat-templates/environments/enable-stf.yaml \
+  --environment-file /home/stack/stf-connectors.yaml \
+  --environment-file /home/stack/publish-to-gnocchi.yaml
+----
+
+After the deployment successfully finished, to verify the configuration was successful, verify the content
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:54]:
```suggestion
. Verification: To verify that the configuration was successful, verify the content
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+of the file `/var/lib/config-data/puppet-generated/ceilometer/etc/ceilometer/pipeline.yaml` on a controller
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:55]:
```suggestion
of the file `/var/lib/config-data/puppet-generated/ceilometer/etc/ceilometer/pipeline.yaml` on a Controller
```

**Comment by mrunge** [comment]
Thank you Joanne, this is very much appreciated.

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+          - "*"
+      sinks:
+          - meter_sink
+sinks:
+    - name: meter_sink
+      publishers:
+          - gnocchi://?filter_project=service&archive_policy=high
+          - notifier://172.17.1.35:5666/?driver=amqp&topic=metering
+----
+Important here is that publishers contain both notifier and gnocchi
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:71]:
```suggestion
+
Ensure that the `publishers` section of the file contains information for both notifier and gnocchi.
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+sections.
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:72]:
```suggestion
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc (hunk 10 above)
Comment (JoanneOFlynn2018) [reply doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:71]: @mrunge Thanks for accepting the other suggestions. This one is minor but just have a quick check to ensure the rewrite hasn't lost anything.

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    CeilometerEnablePanko: false
+    GnocchiArchivePolicy: 'high'
+    GnocchiBackend: 'rbd'
+    GnocchiRbdPoolName: 'metrics'
+
+    EventPipelinePublishers: ['gnocchi://?filter_project=service&archive_policy=high']
+    PipelinePublishers: ['gnocchi://?filter_project=service&archive_policy=high']
+----
+
+. Add the environment file `publish-to-gnocchi.yaml` to the deployment command:
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:40]:
```suggestion
. Add the environment file `publish-to-gnocchi.yaml` to the deployment command. Replace <other-argruments> with files that are applicable to your environment:
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
++
+[source,bash,options="nowrap",subs="+quotes"]
+----
+$ openstack overcloud deploy <other arguments>
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:44]:
```suggestion
$ openstack overcloud deploy <other-arguments>
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 14]
```diff
@@ -0,0 +1,72 @@
+= Sending metrics to Gnocchi and to {ProjectShort}
+
+[role="_abstract"]
+
+To send metrics to {Project} ({ProjectShort}} and gnocchi simultaneously, you must include another configuration file to your deployment. 
+mentioned above.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:6]:
```suggestion
To send metrics to {Project} ({ProjectShort}} and gnocchi simultaneously, you must include an environment file to your deployment to enable an additional publisher.
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+
+.Prerequisites
+
+* You have created a file that contains the connection configuration of the AMQ Interconnect for the overcloud to the {ProjectShort} deployment. For more information, see xref:configuring-the-stf-connection-for-the-overcloud[].
+
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:11]:
```suggestion
* You have created a file that contains the connection configuration of the {MessageBus} for the overcloud to {ProjectShort}. For more information, see xref:configuring-the-stf-connection-for-the-overcloud[].

```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+.Procedure
+
+. Create a configuration file named `publish-to-gnocchi.yaml` in the
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:14]:
```suggestion
. Create an environment file named `gnocchi-connectors.yaml` in the
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+    CeilometerEnablePanko: false
+    GnocchiArchivePolicy: 'high'
+    GnocchiBackend: 'rbd'
+    GnocchiRbdPoolName: 'metrics'
+
+    EventPipelinePublishers: ['gnocchi://?filter_project=service&archive_policy=high']
+    PipelinePublishers: ['gnocchi://?filter_project=service&archive_policy=high']
+----
+
+. Add the environment file `publish-to-gnocchi.yaml` to the deployment command. Replace <other-argruments> with files that are applicable to your environment:
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:40]:
```suggestion
. Add the environment file `gnocchi-connectors.yaml` to the deployment command. Replace <other-argruments> with files that are applicable to your environment:
```

File: doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
++
+[source,bash,options="nowrap",subs="+quotes"]
+----
+$ openstack overcloud deploy <other-arguments>
+  --templates /usr/share/openstack-tripleo-heat-templates \
+  --environment-file <...other-environment-files...> \
+  --environment-file /usr/share/openstack-tripleo-heat-templates/environments/metrics/ceilometer-write-qdr.yaml \
+  --environment-file /usr/share/openstack-tripleo-heat-templates/environments/enable-stf.yaml \
+  --environment-file /home/stack/stf-connectors.yaml \
+  --environment-file /home/stack/publish-to-gnocchi.yaml
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_configuring-gnocchi-metrics.adoc:50]:
```suggestion
  --environment-file /home/stack/gnocchi-connectors.yaml
```

**Review by leifmadsen — APPROVED** [review]
Looks good after a few adjustments for consistency.

**Comment by mrunge** [comment]
Thank you for the reviews and suggestions.

**Review by JoanneOFlynn2018 — APPROVED** [review]

**Pull request merged**